File: day1/main.go
package main

import (
    "fmt"
    "log"
    "os"
    "strconv"
    "strings"
)

func calcSum(X []int) int {
    sum := 0
    for i, _ := range X[1:] {
        if X[i+1] > X[i] {
            sum += 1
        }
    }
    return sum
}

func main() {

    content, err := os.ReadFile("./input")
    if err != nil {
        log.Fatal(err)
    }
    lines := strings.Split(strings.Trim(string(content), "\n"), "\n")

    nums := make([]int, len(lines))
    for i, n := range lines {
        nums[i], _ = strconv.Atoi(n)
    }

    sum := calcSum(nums)
    fmt.Println(sum)

    nums3 := make([]int, len(nums))
    for i, _ := range nums[2:] {
        nums3[i] = nums[i] + nums[i+1] + nums[i+2]
    }

    sum3 := calcSum(nums3)
    fmt.Println(sum3)
}

File: day1/main.py
from pathlib import Path

nums = [int(x) for x in Path("./input").read_text().rstrip().split("\n")]
count = sum(x2 > x1 for x1, x2 in zip(nums, nums[1:]))

print(count)

nums3 = [sum([x1, x2, x3]) for x1, x2, x3 in zip(nums, nums[1:], nums[2:])]
count = sum(x2 > x1 for x1, x2 in zip(nums3, nums3[1:]))

print(count)
